src/serac/numerics/functional/domain_integral.hpp

```diff
@@ -141,17 +137,24 @@ class DomainIntegral {
    * @param[out] output_E The output of the evalution; per-element DOF residuals
    * @see evaluation_kernel
    */
-  void Mult(const mfem::Vector& input_E, mfem::Vector& output_E) const { evaluation_(input_E, output_E); }
+  void Mult(const std::array<mfem::Vector, num_trial_spaces>& input_E, mfem::Vector& output_E, int which = 0) const
+  {
+    if (which == -1) {
+      evaluation_(input_E, output_E);
+    } else {
+      evaluation_with_AD_[which](input_E, output_E);
+    }
+  }
 
   /**
    * @brief Applies the integral, i.e., @a output_E = gradient( @a input_E )
    * @param[in] input_E The input to the evaluation; per-element DOF values
    * @param[out] output_E The output of the evalution; per-element DOF residuals
    * @see gradient_kernel
    */
-  void GradientMult(const mfem::Vector& input_E, mfem::Vector& output_E) const
+  void GradientMult(const mfem::Vector& input_E, mfem::Vector& output_E, size_t which = 0) const
```

Comment:
It might be more explicit (although more verbose) to change `which` to `which_trial` or `which_trial_space`.